Prepare code structure for updated protocol

djordjekovac opened the pull request:
- Refactored publish routes, code moved to commands
- Added new folder/file structure
- Refactored auto-updater module

djordjekovac committed:
updated publish, provision, update routes

djordjekovac committed:
Merge branch 'v6/develop' into v6/refactor/prepare-code-structure-for-updated-protocol

# Conflicts:
#	ot-node.js
#	package-lock.json

zeroxbt committed:
refactor autoupdate-module

djordjekovac committed:
Resolved issues with publish method

djordjekovac committed:
Publish routes refactored

djordjekovac committed:
Updated read from file method in file service

zeroxbt committed:
add symlinks for safer updates

zeroxbt committed:
bump version for testing purposes

zeroxbt committed:
bump version for testing purposes

zeroxbt committed:
fix paths to files and folders

zeroxbt committed:
bump version for testing purposes

zeroxbt committed:
bump version for testing purposes

zeroxbt committed:
fix symlink creation

zeroxbt committed:
fix paths to symlink and folders

zeroxbt committed:
improve npm message logging

zeroxbt committed:
bum version for testing purposes

zeroxbt committed:
remove old version only if not named ot-node

zeroxbt committed:
bump version for testing purposes

zeroxbt committed:
modify comments

zeroxbt committed:
change version back to current (1.35)

NZT48 commented on modules/controller/rpc-controller.js:
```diff
@@ -91,36 +98,41 @@ class RpcController {
     enableSSL() {
-        this.sslEnabled = fs.existsSync('/root/certs/privkey.pem') && fs.existsSync('/root/certs/fullchain.pem');
+        this.sslEnabled =
+            fs.existsSync('/root/certs/privkey.pem') && fs.existsSync('/root/certs/fullchain.pem');
```
Should we move this paths to the constants or maybe to the config?

djordjekovac replied:
I agree, but this is still old controller, for the new one we will move it to config probably. These lines are not changed they are just formated

NZT48 commented on modules/constants.js:
```diff
@@ -263,3 +267,20 @@ exports.ERROR_TYPE = {
+ * @constant {object} HANDLER_ID_STATUS -
+ *  Possible statuses for handler id
+ */
+exports.HANDLER_ID_STATUS = {
```
Should we also have Finished or successful handler id status?

djordjekovac replied:
These constants are added for new controller, and it's still not finished, once we fully use it we will add all statuses

kotlarmilos commented on src/controller/http-api-router.js:
```diff
@@ -0,0 +1,77 @@
+        await this.initializeListeners();
+    }
+
+    async initializeListeners() {
```
It looks so clear now!

kotlarmilos reviewed (approved):
How would sockets fit into the existing code structure for Service API?

djordjekovac committed:
Added folder structure initial migration

djordjekovac committed:
Run migrations only if testnet or mainnet env

djordjekovac committed:
Logging moved to base migration

zeroxbt committed:
refactor autoupdater

djordjekovac committed:
Updated http module interface,
updated autoupdater module interface,
created base module interface,
updated initialization of all modules

djordjekovac committed:
Configuration for autoupdater updated

djordjekovac committed:
Updated naming, interface to manager

djordjekovac committed:
OT-node udpate procedure and file system updated

djordjekovac committed:
Updated dependency installation steps

djordjekovac committed:
Added log if node is running the latest version

djordjekovac committed:
Node is now updating using zip archive from github

djordjekovac committed:
Added current symlink when installing new update

djordjekovac committed:
Updated current symlink in migration script

zeroxbt committed:
update installer to support new folder structure

zeroxbt committed:
Merge branch 'v6/refactor/auto-update-module' of https://github.com/OriginTrail/ot-node into v6/refactor/auto-update-module

djordjekovac committed:
Resolved issues with publish process

djordjekovac committed:
Merge branch 'v6/refactor/prepare-code-structure-for-updated-protocol' into v6/refactor/auto-update-module

# Conflicts:
#	modules/service/file-service.js
#	src/controller/v1/publish-controller.js

zeroxbt committed:
update noderc filepath

zeroxbt committed:
remove nodejs setup folder

zeroxbt committed:
add update file path

zeroxbt committed:
add absolute path to .env file

zeroxbt committed:
add migrations folder path to file service

zeroxbt committed:
update dependency injection paths to be absolute

zeroxbt committed:
remove unnecessary config declarations

zeroxbt committed:
update rc file path

zeroxbt committed:
add correct app name for rc

zeroxbt committed:
update ot-node download link

djordjekovac committed:
Version bump

zeroxbt committed:
update path for db migrations config

zeroxbt committed:
Merge branch 'v6/refactor/auto-update-module' of https://github.com/OriginTrail/ot-node into v6/refactor/auto-update-module

djordjekovac committed:
Updated npm install command

djordjekovac committed:
version bump

zeroxbt committed:
update paths

zeroxbt committed:
Merge branch 'v6/refactor/auto-update-module' of https://github.com/OriginTrail/ot-node into v6/refactor/auto-update-module

zeroxbt committed:
aupdate current directory

zeroxbt committed:
minor fix to sequelize command

zeroxbt committed:
revert changes and bump version

djordjekovac committed:
Updated otnode.service

djordjekovac committed:
Updated path to configuration for network module initialization

djordjekovac committed:
version bump

zeroxbt committed:
remove copy configuration file

zeroxbt committed:
Merge branch 'v6/refactor/auto-update-module' of https://github.com/OriginTrail/ot-node into v6/refactor/auto-update-module

djordjekovac committed:
Merge pull request #1936 from OriginTrail/v6/refactor/auto-update-module

refactor auto-update module

djordjekovac committed:
Merge branch 'v6/develop' into v6/refactor/prepare-code-structure-for-updated-protocol

# Conflicts:
#	config/config.json
#	package-lock.json
#	package.json

djordjekovac committed:
Removed form-data from dependency

zeroxbt committed:
validate file size only if file exists

djordjekovac committed:
Updated dkg client used in bdd tests

zeroxbt committed:
various fixes to new publish route

zeroxbt committed:
change branch in installer

djordjekovac committed:
Fixed issue when sending assertion through data in request

NZT48 commented on tools/local-network-setup/.bootstrap_origintrail_noderc:
```diff
@@ -6,8 +6,8 @@
       ],
-      "publicKey": "<insert_here>",
-      "privateKey": "<insert_here>"
+      "publicKey": "0xE7b10E1C87A2E827AFF5456a39c67eD5D79b82d8",
```
Let's remove this

zeroxbt committed:
add removal of unattended upgrades

djordjekovac committed:
Fixed publish code for bdd tests, pr comments resolved

djordjekovac committed:
Added graphdb service to unit test action

djordjekovac committed:
Updated migration for the new folder structure

djordjekovac committed:
Version bump

djordjekovac committed:
Updated directory paths for migration

djordjekovac committed:
Updated ensuresymlink for folder

djordjekovac committed:
Added unzipper node module

djordjekovac committed:
Updated initial migration

djordjekovac committed:
Added error handling for service file copy

djordjekovac committed:
Moved service file copy after finalize migration

djordjekovac committed:
Migration bug fixed

djordjekovac committed:
Updated system process copy

djordjekovac committed:
Last try for service file

djordjekovac committed:
Remove system file updated

djordjekovac committed:
Updated initial folder migration

djordjekovac committed:
Service file path updated
